### PR TITLE
[TD-12861]<fix>(query): taoshell crash : coredump for such as "select first(c1)==max(c1) from st"

### PR DESCRIPTION
**Pull request by glzhao89**

### CODE REVIEW

**Review by wangmm0220 — APPROVED** [review]

File: src/client/src/tscSQLParser.c [hunk 1]
```diff
@@ -4665,6 +4665,8 @@ static int32_t validateSQLExprItemArithmeticExpr(SSqlCmd* pCmd, tSqlExpr* pExpr,
   const char* msg1 = "arithmetic expression composed with columns from different tables";
   const char* msg2 = "arithmetic expression composed with functions/columns of different types";
   const char* msg3 = "comparison/logical expression involving string operands is not supported";
+  const char* msg4 = "comparison/logical expression involving fucntion result is not supported";
```
Comment (shenglian-zhou) [inline src/client/src/tscSQLParser.c]: typo? fucntion

File: src/query/src/qSqlParser.c [hunk 2]
```diff
@@ -632,7 +632,7 @@ tSqlExpr *tSqlExprClone(tSqlExpr *pSrc) {
   tVariantAssign(&pExpr->value, &pSrc->value);
 
   //we don't clone paramList now because clone is only used for between/and
-  assert(pSrc->Expr.paramList == NULL);
+  pSrc->Expr.paramList = NULL;
```
Comment (shenglian-zhou) [inline src/query/src/qSqlParser.c]: change assert to assignment?

File: src/client/src/tscSQLParser.c (hunk 1 above)
Comment (glzhao89) [reply src/client/src/tscSQLParser.c]: fixed typo.

File: src/query/src/qSqlParser.c (hunk 2 above)
Comment (glzhao89) [reply src/query/src/qSqlParser.c]: discussed with Wei pan, setting the pExpr->Expr.paramList to NULL(not copying the paramlist pointer)

**Review by shenglian-zhou — APPROVED** [review]

**Pull request merged**